test/e2e_node/container_lifecycle_test.go

```diff
@@ -2415,6 +2415,39 @@ var _ = SIGDescribe(nodefeature.SidecarContainers, "Containers Lifecycle", func(
 			ginkgo.It("should start a regular container", func() {
 				framework.ExpectNoError(results.HasRestarted(regular1))
 			})
+
+			ginkgo.It("should not restart when updated with a new image", func() {
+				client := e2epod.NewPodClient(f)
+				podSpec, err := client.Get(context.TODO(), podSpec.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err)
+				results = parseOutput(context.TODO(), f, podSpec)
+
+				tStartRestartableInit1, err := results.TimeOfStart(restartableInit1)
+				framework.ExpectNoError(err)
+				tStartRegular1, err := results.TimeOfStart(regular1)
+				framework.ExpectNoError(err)
+
+				client.Update(context.TODO(), podSpec.Name, func(pod *v1.Pod) {
+					pod.Spec.InitContainers[0].Image = "fakeimage"
+				})
+				framework.ExpectNoError(err)
```

Comment:
The judgment of `err` here is ineffective because the `client.Update` does not return an error.